### DISCUSSION TITLE
Passing multi-key shortcuts to neovim

### DISCUSSION THREAD

**Discussion by joalof** [kovidgoyal/kitty#5693] [Q&A] [answered]
Hi,

I'm trying to pass multi-key shortcuts such as `ctrl+s>j` from kitty to neovim by modifying [this part of the vim-kitty-navigator plugin code](https://github.com/knubie/vim-kitty-navigator/blob/master/pass_keys.py).

The current logic is:
1. Intercept a kitty shortcut
2. Call `parse_shortcut` from  `kitty.key_encoding`
3. Create a KeyEvent (from  `kitty.key_encoding`) using the parsed shortcut
4. Call `window.encoded()` on the event
5. Send to neovim using `window.write_to_child()`

This works well for simple shortcuts such as `ctrl+j` but fails for multi-key maps. Specifically, I would like to pass `ctrl+s>j` from kitty and receive it as `<C+s>j` inside neovim. I've tried to modify the code to accomplish this but I lack so much background knowledge. Is it possible? Do I need to modify the event? Send multiple encoded events? 

Super grateful for any suggestions/hints on how to continue!

**Comment by kovidgoyal** [comment, marked as answer]
you dont need to do anything for that, modern neovim should just support
it out of the box. A multi-key shortcut is just two consecutive
keypresses, which ctrl+s, j is anyway.

**Reply by joalof** [reply]
Thanks for the quick reply! I was able to figure it out now.
